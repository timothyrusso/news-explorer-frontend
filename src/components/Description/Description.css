.description {
  max-width: 600px;
}

.description__title {
  margin: 0;
  padding: 51px 0 24px;
  font-family: 'Roboto Slab', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 40px;
  line-height: 1.15;
}

.description__text {
  margin: 0;
  padding-bottom: 25px;
  font-family: 'Roboto', Arial, sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 1.33;
}

.description__text:last-child {
  padding-bottom: 0;
}

@media screen and (max-width: 1200px) {
  .description {
    margin-left: 30px;
  }
}

@media screen and (max-width: 930px) {
  .description {
    max-width: 424px;
  }

  .description__title {
    font-size: 30px;
    line-height: 1.33;
    padding: 0 0 16px;
  }
}

@media screen and (max-width: 650px) {
  .description {
    display: flex;
    flex-direction: column;
    margin: 0 auto;
  }
}
