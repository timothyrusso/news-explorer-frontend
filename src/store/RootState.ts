import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
