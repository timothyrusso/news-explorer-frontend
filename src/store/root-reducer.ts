import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
