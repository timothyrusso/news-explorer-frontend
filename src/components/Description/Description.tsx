import React from 'react';
import './Description.css';

const Description = () => {
  return (
    <article className="description">
      <h2 className="description__title">About the author</h2>
      <p className="description__text">
        Hey there, I'm Timothy. I'm a motivated and passionate frontend engineer
        with a background in eCommerce, Quality Assurance, Application Delivery
        Management, Project Management and Trade Marketing. I have a bachelor's
        degree in Economics and Business Management and I have done a coding
        bootcamp called TripleTen.
      </p>
      <p className="description__text">
        Iâ€™m a curious person, I love to code and I like to experiment with new
        technologies. Now coding is my passion and my professional career focus.
      </p>
    </article>
  );
};

export default Description;
