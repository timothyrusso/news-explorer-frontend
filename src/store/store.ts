import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
