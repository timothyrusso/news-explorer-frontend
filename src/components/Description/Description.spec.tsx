import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import Description from './Description';

describe('Description component', () => {
  it('Should render the Description component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Description />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
