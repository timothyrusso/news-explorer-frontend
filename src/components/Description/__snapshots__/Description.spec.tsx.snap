// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description component Should render the Description component 1`] = `
<DocumentFragment>
  <article
    class="description"
  >
    <h2
      class="description__title"
    >
      About the author
    </h2>
    <p
      class="description__text"
    >
      Hey there, I'm Timothy. I'm a motivated and passionate frontend engineer with a background in eCommerce, Quality Assurance, Application Delivery Management, Project Management and Trade Marketing. I have a bachelor's degree in Economics and Business Management and I have done a coding bootcamp called TripleTen.
    </p>
    <p
      class="description__text"
    >
      Iâ€™m a curious person, I love to code and I like to experiment with new technologies. Now coding is my passion and my professional career focus.
    </p>
  </article>
</DocumentFragment>
`;
